### DISCUSSION TITLE
Custom document methods returning promise even if function parameter is not async (feature?)

### DISCUSSION THREAD

**Discussion by bros4president** [dynamoose/dynamoose#1189] [Q&A]
Hello! I have defined a custom document instance on an arbitrary `dynamoose.model` object, let's call it `Person`.

An example method could be this:
```
Person.methods.document.set('sayHi', function() {
  return 'hello'
})
````

When I do
```
const person = new Person()
const greeting = person.sayHi()
console.log(greeting)
```

I get `Promise { <pending> }` instead of `hello`.

Now, if I do (notice the await)
```
const person = new Person()
const greeting = await person.sayHi()
console.log(greeting)
```
Nothing gets written on the console.

Finally, if I define the method as async **and** await its execution:
```
Person.methods.document.set('sayHi', async function() {
  return 'hello'
})

const person = new Person()
const greeting = await person.sayHi()
console.log(greeting)
````

I get the desired output `'hello'`, but it looks counterintuitive to me the use of async for a synchronous function or having to use await at execution, knowing it does nothing asyncronous.

So my question is wether this is a feature or if I'm doing something wrong. Thank you :)

**Comment by kevinswarner** [comment]
I noticed the same thing on static methods...

```javascript
Person.methods.set("getFoobar", function () {
  return "foobar";
});
```

The following test fails...

```javascript
expect(Person.getFoobar()).toBe("foobar");
```

The return value is actually a Promise of a string.

Not only that, but if you await in your test, the test will timeout because the promise is not resolved.

If you await in your test, you have to change your function to...

```javascript
Person.methods.set("getFoobar", function () {
  return Promise.resolve("foobar");
});
```

I am just curious about this as well.
